light_curves/light_curve_collector.md

```diff
@@ -155,7 +155,83 @@ or
 You can use [astropy's read](https://docs.astropy.org/en/stable/io/ascii/read.html) function to read in an input table
 to an [astropy table](https://docs.astropy.org/en/stable/table/)
 
-+++
+If you want to build your own sample_table, your table must contain one row per source and include exactly three required pieces of information: a coord column holding an astropy.coordinates.SkyCoord position for each object, a unique integer objectid, and a label column giving a short string describing that source’s origin (e.g., literature reference, sample name, or provenance tag). 
+
+```{code-cell} ipython3
+---
+jupyter:
+  source_hidden: true
+---
+def validate_sample_table(tbl):
+    """
+    Validate the structure of a user-supplied ``sample_table``.
+
+    This function verifies that the supplied object is an
+    ``astropy.table.Table`` and that it contains the required
+    columns (``coord``, ``objectid``, ``label``). It checks that
+    ``coord`` entries are ``SkyCoord`` objects, that ``objectid``
+    values are integers, unique, and sequential starting at 1, and
+    that ``label`` contains strings. Informative exceptions are
+    raised if validation fails.
+    
+    Parameters
+    ----------
+    tbl : astropy.table.Table
+        Table to validate. Must represent a cleaned source sample
+        with one row per sky object.
+
+    Raises
+    ------
+    ValueError
+        If required columns are missing or ``objectid`` values are
+        non-unique or non-sequential.
+    TypeError
+        If column types are not as expected (e.g., ``coord`` not
+        containing ``SkyCoord`` objects).
+
+    """
+    from astropy.coordinates import SkyCoord
+    
+    # Check that tbl is an Astropy Table
+    if not isinstance(tbl, Table):
+        raise TypeError("Input must be an astropy.table.Table instance.")
+
+
+    required = {"coord", "objectid", "label"}
+    missing = required - set(tbl.colnames)
+    if missing:
+        raise ValueError(f"sample_table is missing required columns: {missing}")
+
+    # coord type check
+    if not all(isinstance(c, SkyCoord) for c in tbl["coord"]):
+        raise TypeError("Column 'coord' must contain astropy.coordinates.SkyCoord objects.")
+
+    # objectid checks
+    obj = tbl["objectid"]
+    if not (obj.dtype.kind in ("i", "u")):
+        raise TypeError("Column 'objectid' must contain integers.")
+
+    if len(set(obj)) != len(obj):
+        raise ValueError("Column 'objectid' must contain unique values.")
+
+    if sorted(obj) != list(range(1, len(obj) + 1)):
+        raise ValueError("objectid must start at 1 and increment without gaps.")
+
+    # label check
+    if not all(isinstance(x, str) for x in tbl["label"]):
+        raise TypeError("Column 'label' must contain strings.")
```

Comment:
```suggestion
    # objectid checks
    obj = tbl["objectid"]
    if len(set(obj)) != len(obj):
        raise ValueError("Column 'objectid' must contain unique values.")
```

Having this function is great but I think it's too strict. I don't find anywhere in our code that requires certain types for objectid or label or that requires objectid to be sequential starting at 1. Docstring would need to be updated as well.